internal/command/ssh/sftp.go

```diff
@@ -418,46 +453,64 @@ func (sc *sftpContext) put(args ...string) error {
 		return nil
 	}
 
-	rpath := sc.wd + path.Base(lpath)
-	if rarg := fgs.Arg(1); rarg != "" {
-		if rarg[0] == '/' {
-			rpath = rarg
-		} else {
-			rpath = sc.wd + rarg
-		}
-	}
-
-	if _, err = sc.ftp.Stat(rpath); err == nil {
-		sc.out("put %s -> %s: file exists on VM", lpath, rpath)
-		return nil
-	}
-
-	f, err := os.Open(lpath)
+	matches, err := filepath.Glob(lpath)
 	if err != nil {
-		sc.out("put %s -> %s: open local file: %s", lpath, rpath, err)
+		sc.out("put: match glob %s: %s", lpath, err)
 		return nil
 	}
-	// Safe to ignore the error because this file is for reading.
-	defer f.Close() // skipcq: GO-S2307
 
-	rf, err := sc.ftp.OpenFile(rpath, os.O_WRONLY|os.O_CREATE|os.O_EXCL)
-	if err != nil {
-		sc.out("put %s -> %s: create remote file: %s", lpath, rpath, err)
+	if len(matches) == 0 {
+		sc.out("put: no local files matched the pattern %s", lpath)
 		return nil
 	}
-	defer rf.Close()
 
-	bytes, err := rf.ReadFrom(f)
-	if err != nil {
-		sc.out("put %s -> %s: copy file file: %s (%d bytes written)", lpath, rpath, err, bytes)
-		return nil
-	}
+	rarg := fgs.Arg(1)
 
-	sc.out("%d bytes written", bytes)
+	if len(matches) == 1 {
+		lpath = filepath.ToSlash(matches[0])
 
-	if err = sc.ftp.Chmod(rpath, fs.FileMode(permbits)); err != nil {
-		sc.out("put %s -> %s: set permissions: %s", lpath, rpath, err)
-		return nil
+		rpath := sc.wd + path.Base(lpath)
+		if rarg != "" {
+			if rarg[0] == '/' {
+				rpath = rarg
+			} else {
+				rpath = sc.wd + rarg
+			}
+		}
+
+		sc.out("put %s -> %s: matched 1 file, uploading...", lpath, rpath)
+		sc.uploadFile(lpath, rpath, permbits)
+	} else {
+		rdir := sc.wd
+		if rarg != "" {
+			if rarg[0] == '/' {
+				rdir = rarg
+			} else {
+				rdir = sc.wd + rarg
+			}
+		}
+
+		inf, err := sc.ftp.Stat(rdir)
+		if err != nil {
+			sc.out("put %s -> %s: check remote path: %s", lpath, rdir, err)
+			return nil
+		}
+
+		if !inf.IsDir() {
+			sc.out("put %s -> %s: remote path %s is not a directory", lpath, rdir, rdir)
+			return nil
+		}
+
+		n := len(matches)
+		sc.out("put %s -> %s: matched %d files, uploading...", lpath, rdir, n)
+
+		for i, file := range matches {
+			file = filepath.ToSlash(file)
```

Comment:
had to force `/`s because on Windows, resolved glob patterns are separated by `\`